### PR TITLE
Companion simulator pots and sliders adc refactor fixes

### LINKED ISSUE #3724: Slider not working in companion after ADC refactor
Since ADC refactor, sliders and pots are not longer separated, but slider is now a subtype of POTS, therefore, sliders are currently not working in companion libsimulator

Support for those need to be added.

### PR DESCRIPTION
**Pull request by elecpower**
Fixes #3724

Summary of changes:
- incorporate 3djc PR #3727
- calculate sliders offset in adc
- fix trims credit to @philmoz 
- 6POS calculation using calibration data if exists credit to @philmoz
- fix yaml to convert between pre and post adc refactor

TODO:
- [x] Sliders
- [x] Trims
- [x] 6POS
- [x] Center beep
- [x] Warnings

### CODE REVIEW

File: radio/src/targets/simu/opentxsimulator.cpp [hunk 1]
```diff
@@ -236,16 +239,23 @@ void OpenTxSimulator::setTrainerInput(unsigned int inputNumber, int16_t value)
 
 void OpenTxSimulator::setInputValue(int type, uint8_t index, int16_t value)
 {
-  //qDebug() << type << index << value << this->thread();
+  qDebug() << type << index << value << this->thread();
```
Comment (gagarinlg) [inline radio/src/targets/simu/opentxsimulator.cpp:242]: I think this should be disabled again

Comment (elecpower) [reply radio/src/targets/simu/opentxsimulator.cpp:242]: Agree but still working on trims and there will be more like this until resolved. Changed PR to Merge state so all libsims compile

**Comment by philmoz** [comment]
On my system to get the 6POS to work correctly in the simulator (using calibration data from my radio) requires this code:
```
uint16_t simu_get_analog(uint8_t idx)
{
  // 6POS simu mechanism use a different scale, so needs specific offset
  if (IS_POT_MULTIPOS(idx - adcGetInputOffset(ADC_INPUT_POT)))
    return (g_anas[idx] * 1382 / 2048);
  return (g_anas[idx] * 2) + 2048;
}
```

**Comment by philmoz** [comment]
To fix the simulator trims change line 482 of radio/src/targets/simu/simpgmspace.cpp to:
```
  for (int i = 0; i < MAX_TRIMS*2; i++) {
```

**Comment by elecpower** [comment]
> To fix the simulator trims change line 482 of radio/src/targets/simu/simpgmspace.cpp to:

Congrats you beat me to the source of the issue. I had cross checked the higher code and all appeared okay except that higher indexes for trim up/down trim presses were not being emitted from the radio. So I was about to delve into the radio side. Thanks again.

**Comment by elecpower** [comment]
> On my system to get the 6POS to work correctly in the simulator (using calibration data from my radio) requires this code:
> 
> ```
> uint16_t simu_get_analog(uint8_t idx)
> {
>   // 6POS simu mechanism use a different scale, so needs specific offset
>   if (IS_POT_MULTIPOS(idx - adcGetInputOffset(ADC_INPUT_POT)))
>     return (g_anas[idx] * 1382 / 2048);
>   return (g_anas[idx] * 2) + 2048;
> }
> ```

I'll leave the discussion as to the 'best' formula to yourself, @3djc and others.

**Comment by philmoz** [comment]
Here is a replacement for the simu_get_analog function that calculates the 6POS conversion factor from the radio calibration data in the profile.

This should work in the companion simulator when there is 6POS calibration data loaded from the user's radio. It does not work when there is no valid calibration data for the 6POS switch; but then 2.9 and earlier don't work in this case either.

```
uint16_t simu_get_analog(uint8_t idx)
{
  // 6POS simu mechanism use a different scale, so needs specific offset
  if (IS_POT_MULTIPOS(idx - adcGetInputOffset(ADC_INPUT_POT))) {
    // Use radio calibration data to determine conversion factor
    StepsCalibData * calib = (StepsCalibData *) &g_eeGeneral.calib[idx];
    int range6POS = 2048; // Default if calibration is not valid
    if (calib->count != 0) {
      // calculate 6POS switch range from calibration data
      int c1 = calib->steps[calib->count - 1] * 32; // last calibration value
      int c2 = calib->steps[calib->count - 2] * 32; // 2nd last calibration value
      range6POS = c1 + (c1 - c2) / 2;
    }
    return (g_anas[idx] * range6POS / 2048);
  }
  return (g_anas[idx] * 2) + 2048;
}
```

**Comment by elecpower** [comment]
Tested new model files self compiled main with addition of @3djc fixes for editing B&W radio for Boxer, X9D+ and TX16S:
- Companion write and read
- libsim mirrors Companion
Tested new model files self compiled main for X9D+ and TX16S:
- Companion write and read
- libsim mirrors Companion

**Pull request merged**